alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/alerts/AlertIndices.kt

```diff
@@ -235,12 +260,61 @@ class AlertIndices(
                 .mapping(MAPPING_TYPE, alertMapping(), XContentType.JSON)
         request.addMaxIndexDocsCondition(historyMaxDocs)
         request.addMaxIndexAgeCondition(historyMaxAge)
-        val response = client.rolloversIndex(request).actionGet(requestTimeout)
+        val response = client.admin().indices().rolloversIndex(request).actionGet(requestTimeout)
         if (!response.isRolledOver) {
             logger.info("$HISTORY_WRITE_INDEX not rolled over. Conditions were: ${response.conditionStatus}")
         } else {
             lastRolloverTime = TimeValue.timeValueMillis(threadPool.absoluteTimeInMillis())
         }
         return response.isRolledOver
     }
+
+    private fun deleteOldHistoryIndices() {
+        val indicesToDelete = mutableListOf<String>()
+
+        val clusterStateRequest = ClusterStateRequest()
+                .clear()
+                .indices(HISTORY_ALL)
+                .metaData(true)
+                .local(true)
```

Comment:
doesn't this mean you are only getting local indices rather than cluster wide indices? I don't recall if this job runs on all nodes, or only on a single one.

Reply:
Right, using ```local(true)``` will have it run on master in this case, which is the behavior we want here.